src/query/v2/plan/operator.cpp

```diff
@@ -462,6 +463,202 @@ class DistributedScanAllAndFilterCursor : public Cursor {
   std::optional<std::vector<Expression *>> filter_expressions_;
 };
 
+class DistributedScanAllByPrimaryKeyCursor : public Cursor {
+ public:
+  explicit DistributedScanAllByPrimaryKeyCursor(
+      Symbol output_symbol, UniqueCursorPtr input_cursor, const char *op_name,
+      std::optional<storage::v3::LabelId> label,
+      std::optional<std::pair<storage::v3::PropertyId, Expression *>> property_expression_pair,
+      std::optional<std::vector<Expression *>> filter_expressions, std::optional<std::vector<Expression *>> primary_key)
+      : output_symbol_(output_symbol),
+        input_cursor_(std::move(input_cursor)),
+        op_name_(op_name),
+        label_(label),
+        property_expression_pair_(property_expression_pair),
+        filter_expressions_(filter_expressions),
+        primary_key_(primary_key) {
+    ResetExecutionState();
+  }
+
+  enum class State : int8_t { INITIALIZING, COMPLETED };
+
+  using VertexAccessor = accessors::VertexAccessor;
+
+  bool MakeRequest(RequestRouterInterface &request_router, ExecutionContext &context) {
```

Comment:
Why do we have `MakeRequest` and `MakeRequestSingleFrame`, when is one called and when the other?

Reply:
Atm, we have to set a flag depending on what frame-type we want to use in the pull-chain, however the multi-frame version is not implemented in this PR, only the underlying logic that makes it possible to use and pick the new operator when appropriate and the single-frame version of its execution.
In order to get the `PrimaryKey` we have to evaluate the `Expression*`s that are members of the cursor object. For that we need to pass the `Frame&` to the `MakeRequestSingleFrame` function. Once the multi-frame version is implemented, it will probably take `MultiFrame&` to evaluate the same stuff and once we settle the `MultiFrame` vs `Frame` discussion one of these functions will disappear.
Since the currently implemented `MultiFrame` operators use the `MakeRequest` name, I did not wanted to use that in the case of the `SingleFrame` solution.
So,
single-frame solution -> `MakeRequestSingleFrame`
multi-frame solution -> `MakeRequest`

As for the multi-frame solution I will mark it not implemented appropriately, I forgot to do that.

Reply:
Aha ok thanks for explanation :+1: